src/Core/Resolvers/SqlPaginationUtil.cs

```diff
@@ -102,7 +102,9 @@ private static JsonObject CreatePaginationConnection(JsonElement root, Paginatio
             {
                 // parse *Connection.endCursor if there are no elements
                 // if no after is added, but it has been requested HotChocolate will report it as null
-                if (returnedElemNo > 0)
+                // Need to validate we have an extra element, because otherwise there is no next page
+                // and endCursor should be left as null.
+                if (returnedElemNo > 0 && hasExtraElement)
```

Comment:
This conditional introduces a regression that also needs an added test to protect against once you identify a fix:

This query works as expected in `main` but fails with this change:

`main`
```graphql
query GetBooksPaged {
  books(first: 5) {
    items {
      id
      title
      publisher_id
    }
    endCursor
  }
}
```
result which includes endCursor
``` json
{
  "data": {
    "books": {
      "items": [
        { "<items>":"<here>"}
      ],
      "endCursor": "W3siRW50aXR5TmFtZSI6IkJvb2siLCJGaWVsZE5hbWUiOiJpZCIsIkZpZWxkVmFsdWUiOjUsIkRpcmVjdGlvbiI6MH1d"
    }
  }
}
```

However, with this PR's changes,
the same query has a `null` for `endCursor`. Weird, because including the field `hasNextPage` results in the `endCursor` being properly supplied (non-null)

``` json
{
  "data": {
    "books": {
      "items": [
        { "<items>":"<here>"}
      ],
      "endCursor": null
    }
  }
}
```

This works fine when `hasNextPage` field is included .. which is odd:
```graphql
query GetBooksPaged {
  books(first: 5) {
    items {
      id
      title
      publisher_id
    }
    endCursor
    hasNextPage
  }
}
```
result:
```json
{
  "data": {
    "books": {
      "items": [
        { "<items>":"<here>"}
      ],
      "endCursor": "W3siRW50aXR5TmFtZSI6IkJvb2siLCJGaWVsZE5hbWUiOiJpZCIsIkZpZWxkVmFsdWUiOjUsIkRpcmVjdGlvbiI6MH1d",
      "hasNextPage": true
    }
  }
}
```

Reply:
it is because the check for pagination (looking at if we have extra elements after the current page) that sets the value for `hasExtraElement` was hidden inside of a check looking for if we have included `hasNextPage`, I've moved the setting of this value outside of that conditional which should fix everything.

Reply:
> This conditional introduces a regression that also needs an added test to protect against once you identify a fix:
> 
> This query works as expected in `main` but fails with this change:


More context for what changed, and how we are handling this.

Main is handling this specific query correctly, but only because it happens to fulfill a scenario where the behavior is correct, not because we have the correct logic for handling pagination with only endCursor in the request.

This is because we would include an endCursor anytime there were elements returned greater than 0, and so in this case, that is met, and the correct endCursor comes back. However, we could have the case where more than 0 elements are returned by an endCursor should not be included.

To properly handle these cases in their entirety, we need our logic for correctly handling pagination scenarios to extend to requests that include either endCursor, hasNextPage, or both.

I have made changes to accommodate this and will add the details to the PR description.

Reply:
> However, we could have the case where more than 0 elements are returned by an endCursor should not be included.

Can you identify any examples of this? would be good to include in pr description to illustrate your findings.